src/org/javarosa/xform/parse/StandardBindAttributesProcessor.java

```diff
@@ -48,39 +54,21 @@ DataBinding createBinding(IXFormParserFunctions parserFunctions, FormDef formDef
 
         binding.setDataType(getDataType(element.getAttributeValue(null, "type")));
 
-        final String xpathRel = element.getAttributeValue(null, "relevant");
-        if (xpathRel != null) {
-            if ("true()".equals(xpathRel)) {
-                binding.relevantAbsolute = true;
-            } else if ("false()".equals(xpathRel)) {
-                binding.relevantAbsolute = false;
-            } else {
-                binding.relevancyCondition = getCondition(formDef, ref, xpathRel, "relevant");
-            }
-        }
-
-        final String xpathReq = element.getAttributeValue(null, "required");
-        if (xpathReq != null) {
-            if ("true()".equals(xpathReq)) {
-                binding.requiredAbsolute = true;
-            } else if ("false()".equals(xpathReq)) {
-                binding.requiredAbsolute = false;
-            } else {
-                binding.requiredCondition = getCondition(formDef, ref, xpathReq, "required");
-            }
-        }
-
-        final String xpathRO = element.getAttributeValue(null, "readonly");
-        if (xpathRO != null) {
-            if ("true()".equals(xpathRO)) {
-                binding.readonlyAbsolute = true;
-            } else if ("false()".equals(xpathRO)) {
-                binding.readonlyAbsolute = false;
-            } else {
-                binding.readonlyCondition = getCondition(formDef, ref, xpathRO, "readonly");
-            }
-        }
-
+        setBoolAndCondition(ref, element, "relevant", formDef, new BoolAndConditionSetter() {
```

Comment:
I think my experience with functional languages led me here. This bit would look really clean in Scala. The point was to remove the code duplication among relevant, required, and readonly. Maybe improving the names would make it more readable? 

In any case, I’m happy to put it back as it was, if you like.